File: generator-common/src/main/java/io/github/potjerodekool/openapi/common/dependency/ConditionalOnDependency.java
package io.github.potjerodekool.openapi.common.dependency;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE, ElementType.METHOD})
@Conditional
public @interface ConditionalOnDependency {

    String groupId();
    String artifactId();

}

File: generator-common/src/main/kotlin/io/github/potjerodekool/openapi/common/dependency/ConditionalOnDependency.kt
package io.github.potjerodekool.openapi.common.dependency

@Retention(AnnotationRetention.RUNTIME)
@Target(
    AnnotationTarget.CLASS,
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER
)
@Conditional
annotation class ConditionalOnDependency(val groupId: String, val artifactId: String)
